airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/write/transform/SnowflakeValueCoercer.kt

```diff
@@ -62,10 +64,13 @@ fun isVarcharValid(s: String): Boolean {
 }
 
 @Singleton
-class SnowflakeValueCoercer : ValueCoercer {
+class SnowflakeValueCoercer(val config: SnowflakeConfiguration) : ValueCoercer {
     override fun map(value: EnrichedAirbyteValue): EnrichedAirbyteValue {
         value.abValue =
-            if (value.type is UnionType) {
+            if (
+                !config.legacyRawTablesOnly &&
+                    (value.type is UnionType || value.type is UnknownType)
+            ) {
```

Comment:
this is the correct way to write variant values successfully

Reply:
@tryangul what's your opinion on this? i.e. using `ValueCoercer.map` to handle UnknownType in addition to UnionType

the alternative here is that SnowflakeAggregate needs to check whether a field is union vs unknown (since they both map to snowflake `VARIANT`) before maybe calling `serializeToString`. Which feels wrong to me

Reply:
I think that's fine